File: trunk/src/main/java/net/esper/timer/EQLTimerTask.java
package net.esper.timer;

import java.util.TimerTask;

/**
 * Timer task to simply invoke the callback when triggered.
 */
final class EQLTimerTask extends TimerTask
{
    private final TimerCallback callback;

    public EQLTimerTask(TimerCallback callback)
    {
        this.callback = callback;
    }

    public final void run()
    {
        callback.timerCallback();
    }
}

File: NEsper/timer/EQLTimerTask.cs
using System;
using System.Timers;

namespace net.esper.timer
{
    /// <summary>
    /// Timer task to simply invoke the callback when triggered.
    /// </summary>

    sealed class EQLTimerTask
    {
        private readonly TimerCallback timerCallback;
        private bool isCancelled;

        public bool Cancelled
        {
            set { isCancelled = value; }
        }

        public EQLTimerTask(TimerCallback callback)
        {
            this.timerCallback = callback;
        }

        public void Run( object sender, ElapsedEventArgs e)
        {
            if (!isCancelled)
            {
                timerCallback();
            }
        }
    }
}